packages/connect/src/create-store.js

```diff
@@ -21,12 +21,18 @@ const getSnapshot = obj => {
 
 const convertToAction = (fn, instance) => (...args) => {
   const first = fn(instance);
-  if (first instanceof Promise)
-    return new Promise(resolve => first.then(() => resolve()));
+  if (first instanceof Promise) {
+    return new Promise((resolve, reject) =>
+      first.then(resolve).catch(err => reject(err))
+    );
+  }
   if (typeof first === "function") {
     const second = first(...args);
-    if (second instanceof Promise)
-      return new Promise(resolve => second.then(() => resolve()));
+    if (second instanceof Promise) {
+      return new Promise((resolve, reject) =>
+        second.then(resolve).catch(err => reject(err))
```

Comment:
Same here, don't do `then(resolve)` but `then(() => resolve)` to avoid returning the value.